packages/components/inputs/src/base-checkbox/types.ts

```diff
@@ -0,0 +1,37 @@
+import type { ComponentProps } from 'react';
+import type { BaseInputInternalProps } from '../base-input/types';
+
+export type checkboxTypes = 'checkbox' | 'radio' | 'switch';
+
+export interface BaseCheckboxProps
+  extends Omit<
+    BaseInputInternalProps,
+    'type' | 'as' | 'placeholder' | 'isReadOnly' | 'icon' | 'isInvalid'
```

Comment:
I'm not sure if we need to exclude `isInvalid` here

Reply:
yeah, I think we actually gonna need it.